Automattic/pocket-casts-ios
# Add a helper extension to make testing themes and devices easier with SwiftUI

This takes advantage of the SwiftUI PreviewProviders ability to show multiple previews in the same view, by adding some helper methods to:
- Add previews for all the themes in PC
- Add previews for specific devices and themes without needing to recompile
- Add previews for our common testing size devices (iPhone SE, iPhone 14 Pro Max, and iPhone 8)

Please include a summary of the changes and the related issue.

## To test

1. Open any SwiftUI `PreviewProvider` that takes the Theme as an environment variable
2. Replace the existing variable with `.preview(with: .light)` 
3. ✅ Verify the preview is generated with current theme
4. Change the theme to something else and ✅ Verify it changes
5. Change the .preview with `.previewWithAllThemes()`
6. ✅ Verify the preview is updated and you see all the theme names in the top 
7. Click on the themes and verify they are correct
8. Change .previewWithAllThemes to `previewOnAllDevices`
9. ✅ Verify the preview is updated and shows The current device + 3 more
10. Verify clicking on them updates the preview to that device

## Checklist
